keras_cv/src/models/object_detection/faster_rcnn/faster_rcnn.py

```diff
@@ -0,0 +1,653 @@
+                "`loss` will be ignored during training."
+            )
+        rpn_box_loss = _parse_box_loss(rpn_box_loss)
+        rpn_classification_loss = _parse_rpn_classification_loss(
+            rpn_classification_loss
+        )
+
+        if hasattr(rpn_classification_loss, "from_logits"):
+            if not rpn_classification_loss.from_logits:
+                raise ValueError(
+                    "FasterRCNN.compile() expects `from_logits` to be True for "
+                    "`rpn_classification_loss`. Got "
+                    "`rpn_classification_loss.from_logits="
+                    f"{rpn_classification_loss.from_logits}`"
+                )
+        box_loss = _parse_box_loss(box_loss)
+        classification_loss = _parse_classification_loss(classification_loss)
+
+        if hasattr(classification_loss, "from_logits"):
+            if not classification_loss.from_logits:
+                raise ValueError(
+                    "FasterRCNN.compile() expects `from_logits` to be True for "
+                    "`classification_loss`. Got "
+                    "`classification_loss.from_logits="
+                    f"{classification_loss.from_logits}`"
+                )
+        if hasattr(box_loss, "bounding_box_format"):
+            if box_loss.bounding_box_format != self.bounding_box_format:
+                raise ValueError(
+                    "Wrong `bounding_box_format` passed to `box_loss` in "
+                    "`FasterRCNN.compile()`. Got "
+                    "`box_loss.bounding_box_format="
+                    f"{box_loss.bounding_box_format}`, want "
+                    "`box_loss.bounding_box_format="
+                    f"{self.bounding_box_format}`"
+                )
+
+        self.rpn_box_loss = rpn_box_loss
+        self.rpn_cls_loss = rpn_classification_loss
+        self.box_loss = box_loss
+        self.cls_loss = classification_loss
+        self.weight_decay = weight_decay
+        losses = {
+            "rpn_box": self.rpn_box_loss,
+            "rpn_classification": self.rpn_cls_loss,
+            "box": self.box_loss,
+            "classification": self.cls_loss,
+        }
+        self._has_user_metrics = metrics is not None and len(metrics) != 0
+        self._user_metrics = metrics
+        super().compile(loss=losses, **kwargs)
+
+    def compute_loss(
+        self, x, y, y_pred, sample_weight, training=True, **kwargs
+    ):
+
+        # 1. Unpack the inputs
+        images = x
+        gt_boxes = y["boxes"]
+        if ops.ndim(y["classes"]) != 2:
+            raise ValueError(
+                "Expected 'classes' to be a Tensor of rank 2. "
+                f"Got y['classes'].shape={ops.shape(y['classes'])}."
+            )
+
+        gt_classes = y["classes"]
+        gt_classes = ops.expand_dims(gt_classes, axis=-1)
+
+        #######################################################################
+        # Generate  Anchors and Generate RPN Targets
```

Comment:
lets keep the comment here simple without the `#####`

Reply:
Done